tools/cldr-code/src/main/java/org/unicode/cldr/util/StandardCodes.java

```diff
@@ -84,6 +84,26 @@ public NameType toNameType() {
                     return NameType.NONE;
             }
         }
+
```

Comment:
Better to make the related enum values be elements of the other one.

That is, CodeType would have a field set from its private constructor:

```
final NameType nameType;
```

Then toNameType becomes:

```
public NameType toNameType() {
  return nameType;
}
```
Similarly, NameType would have a field set from its private constructor:

```
final CodeType codeType;
```
Then fromNameType moves to be a method NameType.toCodeType.

```
public CodeType toNameType() {
  if (codeType == null) {
    throw new IllegalArgumentException("Unsupported name type");
  }
  return codeType;
}
```

Reply:
good idea, I'll make that change

Reply:
@macchiati I thought it was a great idea, but it failed due to the sequence of initialization. If NameType constructor references CodeType, and CodeType constructor references NameType, one or the other will get null values.

See https://stackoverflow.com/questions/1506410/java-enums-two-enum-types-each-containing-references-to-each-other which suggestions various workarounds, but they're all kind of ugly.

Instead I just made it a little more symmetrical by implementing NameType.toCodeType instead of CodeType.fromNameType.

Maybe there's still a more elegant or efficient way; I suggest leaving that for a follow-up PR if and when it seems worthwhile to pursue.